src/data/gateway-data-source.ts

```diff
@@ -42,18 +47,36 @@ export class GatewayDataSource implements ContiguousDataSource {
     });
   }
 
-  async getData(id: string): Promise<ContiguousData> {
+  async getData({
+    id,
+    requestAttributes,
+  }: {
+    id: string;
+    requestAttributes?: RequestAttributes;
+  }): Promise<ContiguousData> {
     const path = `/raw/${id}`;
     this.log.info('Fetching contiguous data from gateway', {
       id,
       trustedGatewayUrl: this.trustedGatewayAxios.defaults.baseURL,
       path,
     });
 
+    const reqOriginAndHopsHeaders: { [key: string]: string } = {};
+    const reqHops = requestAttributes?.hops ?? 0;
```

Comment:
I believe in order for hops to increase as requests propagate around the network (in the future) we need to increment any hops number we received on the inbound request. I also wonder as I look at the number of times we're incrementing when we return `requestAttributes` whether defaulting to `1` on outbound requests instead of `0` would simplify the code.